src/serverless/functions/recordingrules.js

```diff
@@ -0,0 +1,51 @@
+/* global Twilio */
+'use strict';
+
+// We need to use a newer twilio client than the one provided by context.getTwilioClient(),
+// so we require it here. The version is specified in helpers.js in the 'deployOptions' object.
+const twilio = require('twilio');
+
+module.exports.handler = async (context, event, callback) => {
+  const authHandler = require(Runtime.getAssets()['/auth-handler.js'].path);
+  authHandler(context, event, callback);
```

Comment:
I know we talked about this same thing [here](https://github.com/twilio/rtc-diagnostics-react-app/pull/31#discussion_r529788476), but I like to bring it up again and maybe we should think more about this pattern. We are somehow trying to do a middleware paradigm but how we're using the middleware is not ideal. For example in this implementation, we have to require each middleware in each endpoints. And if we add more middleware, or any preprocessor for each endpoint, this could get ugly.

Ideally, we should use a generic approach to each of our endpoints. For example, this handler could look something like below:
```js
const { useMiddlewares } = require('/middlewares');

module.exports.handler = useMiddlewares(async (context, event, callback) => {
  let response = new Twilio.Response();
  response.appendHeader('Content-Type', 'application/json');
  // rest of the handler logic...
});
```

With this approach, you only require `useMiddlewares` in each endpoint. `useMiddlewares` module will then handle loading all middlewares, including the auth middleware. Personally, this feels cleaner and easier to maintain.

Reply:
For the record, Charlie and I discussed this one in person, and we agreed to make this change as part of this ticket: https://issues.corp.twilio.com/browse/AHOYAPPS-963

Reply:
I agree with this change. Thanks for the detailed comment 👍🏼